wiki/Introduction-to-Creating-a-Job.md

```diff
@@ -0,0 +1,1818 @@
+    script:
+      actions:
+        onRun:
+          command: "{{Task.File.Render}}"
+          args:
+            - "{{Param.SceneFile}}"
+            - "{{Param.FramesDirectory}}"
+            - "{{Task.Param.Frames}}"
+          timeout: 2400 # 40 frames at 60s per frame
+      embeddedFiles:
+        - name: Render
+          type: TEXT
+          filename: render.sh
+          runnable: true
+          data: |
+            #!/bin/bash
+
+            # Return an error code if any command in the script fails.
+            set -eou pipefail
+
+            TMPFILE=$(mktemp)
+            cat > "$TMPFILE" << EOF
+            import bpy
+            for s in bpy.data.scenes:
+              s.render.resolution_x = 480
+              s.render.resolution_y = 270
+            bpy.context.scene.cycles.samples = 100
+            EOF
+
+            SCENE="$1"
+            OUTDIR="$2"
+            FRAME="$3"
+
+            blender --background "$SCENE" \
+              --python "$TMPFILE" \
+              --render-output "$OUTDIR"/frame-### \
+              --render-format PNG --use-extension 1 \
+              --render-frame "${FRAME}"
+
+            rm -f $TMPFILE
+
+  - name: EncodeVideo
+    dependencies:
+      - dependsOn: BlenderRender
+    script:
+      actions:
+        onRun:
+          command: "{{Task.File.Encode}}"
+          args:
+            - "{{Param.FramesDirectory}}"
+            - "{{Param.AnimationFile}}"
+            - "1"
+          timeout: 60
+      embeddedFiles:
+        - name: Encode
+          type: TEXT
+          runnable: true
+          filename: encode.sh
+          data: |
+            #!/bin/bash
+
+            set -eou pipefail
+
+            INPUT_DIR="$1"
+            OUTPUT_FILENAME="$2"
+            START_FRAME="$3"
+
+            ffmpeg -y -r 10 -start_number "$START_FRAME" -i "$INPUT_DIR"/frame-%03d.png -pix_fmt yuv420p \
+                -vf "scale=in_color_matrix=bt709:out_color_matrix=bt709" \
+                -frames:v 300 -c:v libx264 -preset fast \
+                -color_range tv -colorspace bt709 -color_primaries bt709 -color_trc iec61966-2-1 \
+                -movflags faststart "$OUTPUT_FILENAME"
+```
+
+As usual, you can check the syntax of the template and then try it out locally using the CLI:
+
+```bash
+% openjd check job.template.yaml                     
+Template at 'job.template.yaml' passes validation checks.
+
+% openjd run --step BlenderRender job.template.yaml \
+  -tp Frames=1..40 \
+  -p SceneFile=$(pwd)/3d/pavillon_barcelone_v1.2.blend \
+  -p FramesDirectory=$(pwd)/output_frames \
+  -p AnimationFile=$(pwd)/animation.mp4
+Mon Jul  8 10:48:30 2024	
+Mon Jul  8 10:48:30 2024	==============================================
+Mon Jul  8 10:48:30 2024	--------- Running Task
+Mon Jul  8 10:48:30 2024	==============================================
+Mon Jul  8 10:48:30 2024	Parameter values:
+Mon Jul  8 10:48:30 2024	Frames(STRING) = 1..40
+...
+```
+
+### 2.3. Ready for Production
```

Comment:
Note to self: We should add something on the `hostRequirements` property in here.